src/servers/src/postgres/handler.rs

```diff
@@ -61,36 +61,43 @@ impl SimpleQueryHandler for PostgresServerHandler {
         C: ClientInfo + Unpin + Send + Sync,
     {
         let query_ctx = query_context_from_client_info(client);
-        let output = self
-            .query_handler
-            .do_query(query, query_ctx)
-            .await
-            .map_err(|e| PgWireError::ApiError(Box::new(e)))?;
+        let outputs = self.query_handler.do_query(query, query_ctx).await;
 
-        match output {
-            Output::AffectedRows(rows) => Ok(vec![Response::Execution(Tag::new_for_execution(
-                "OK",
-                Some(rows),
-            ))]),
-            Output::Stream(record_stream) => {
-                let schema = record_stream.schema();
-                recordbatches_to_query_response(record_stream, schema)
-            }
-            Output::RecordBatches(recordbatches) => {
-                let schema = recordbatches.schema();
-                recordbatches_to_query_response(
-                    stream::iter(recordbatches.take().into_iter().map(Ok)),
-                    schema,
-                )
-            }
+        let mut results = Vec::with_capacity(outputs.len());
+
+        for output in outputs {
+            let resp = match output {
+                Ok(Output::AffectedRows(rows)) => {
+                    Response::Execution(Tag::new_for_execution("OK", Some(rows)))
+                }
+                Ok(Output::Stream(record_stream)) => {
+                    let schema = record_stream.schema();
+                    recordbatches_to_query_response(record_stream, schema)?
+                }
+                Ok(Output::RecordBatches(recordbatches)) => {
+                    let schema = recordbatches.schema();
+                    recordbatches_to_query_response(
+                        stream::iter(recordbatches.take().into_iter().map(Ok)),
+                        schema,
+                    )?
+                }
+                Err(e) => Response::Error(Box::new(ErrorInfo::new(
+                    "ERROR".to_string(),
+                    "XX000".to_string(),
```

Comment:
Should we assign a code for it?

Reply:
this is following postgresql's error code schema: https://www.postgresql.org/docs/current/errcodes-appendix.html

At the moment we treat them all as internal error, but we can further map our status codes to postgres error codes in future.